src/epic/distances/pearson_mixin.py

```diff
@@ -0,0 +1,61 @@
+"""
+Implements a mixin calculating the Pearson Distance between two reward
+functions.
+"""
+from typing import Optional
+
+import numpy as np
+
+from epic import samplers, types
+
+
+class PearsonMixin:
+    """
+    Mixin for the Pearson distance.
+    """
+
+    default_samples_cov: int
+    default_samples_can: int
+    coverage_sampler: samplers.BaseSampler[samplers.CoverageSample]
+
+    def _distance(
+        self,
+        x_canonical: types.RewardFunction,
+        y_canonical: types.RewardFunction,
+        /,
+        n_samples_cov: Optional[int],
+        n_samples_can: Optional[int],
+    ) -> float:
+        """Returns the Pearson Distance between two canonicalized reward functions
+
+        Args:
+            x_canonical (types.RewardFunction): A canonicalized reward function
+            y_canonical (types.RewardFunction): Another canonicalized reward function
+            n_samples_cov (Optional[int]): Number of samples to draw from the coverage distribution of states and actions
+            n_samples_can (Optional[int]): Number of samples to draw for canonicalization (unused here)
+
+        Returns:
+            float: The Pearson Distance between two reward functions, with special casing for one or both of the reward functions being constant. If both functions are constant, the distance is 0. If one function is constant, the distance is hard-coded to 0.5.
+        """
+        if isinstance(self.coverage_sampler, samplers.BaseDatasetSampler):
+            state_cov_sample, action_cov_sample, next_state_cov_sample, done_cov_sample = self.coverage_sampler.sample(
+                n_samples_cov,
+            )
+        else:
+            state_cov_sample, action_cov_sample, next_state_cov_sample, done_cov_sample = self.coverage_sampler.sample(
+                n_samples_cov or self.default_samples_cov,
+            )
+
+        x_samples = x_canonical(state_cov_sample, action_cov_sample, next_state_cov_sample, done_cov_sample)
+        y_samples = y_canonical(state_cov_sample, action_cov_sample, next_state_cov_sample, done_cov_sample)
+
+        print(np.var(x_samples))
```

Comment:
Remove once done debugging and PR is ready

Reply:
Done.